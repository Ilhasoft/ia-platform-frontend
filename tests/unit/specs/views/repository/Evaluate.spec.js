import { shallowMount, createLocalVue } from '@vue/test-utils';
import EvaluateManual from '@/views/repository/EvaluateManual';

import Vuex from 'vuex';
import Buefy from 'buefy';

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(Vuex);

describe('EvaluateManual.vue', () => {
  let wrapper;
  let state;
  let store;
  let actions;
  let getters;
  beforeEach(() => {
    state = {
      selectedRepository: {
        uuid: 'dbe1539c-88ba-43e0-bed5-54729218cec9',
      },
    };
    actions = {
      getEvaluateExample: jest.fn(),
      runNewEvaluate: jest.fn(),
      setUpdateRepository: jest.fn(),
      setRepositoryVersion: jest.fn(),
    };
    getters = {
      authenticated: jest.fn(),
      getUpdateRepositoryState: jest.fn(),
      getSelectedVersion: jest.fn(),
      activeTutorial: jest.fn()

    };
    store = new Vuex.Store({
      modules: {
        Repository: {
          state,
          getters,
          actions,
        },
      },
    });
    wrapper = shallowMount(EvaluateManual, {
      localVue,
      store,
      mocks: {
        $t: () => 'some specific text',
        $route: {
          name: 'repository-result',
          params: {
            ownerNickname: 'teste-pubngj',
            slug: 'teste-privado',
            resultId: 3525,
            version: 2,
          },
        }
      }
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('run new test', () => {
    expect(wrapper.vm.newEvaluate()).toBeTruthy();
  });
});


