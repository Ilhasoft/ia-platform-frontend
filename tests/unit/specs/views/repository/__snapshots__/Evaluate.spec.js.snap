// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluateManual.vue renders correctly 1`] = `
<repository-view-base-stub repository="<Repository #1>" quicktest="true">
  <div>
    <div>
      <b-notification-stub active="true" type="is-info" duration="2000" animation="fade">
        some specific text
      </b-notification-stub>
      <login-form-stub hideforgotpassword="true"></login-form-stub>
    </div>
  </div>
  <!---->
</repository-view-base-stub>
`;
